### PR TITLE
Application:HUST_SoundSnake

### PR DESCRIPTION
**Pull request by hustmzp**
# Summary
- embarc_application:HUST_SoundSnake.
-  The development has been completed.

# ToDos
- Nothing to do further.

# User Manual
- Please find the User Manual in this application's README file.

## NOTES: The link script has been modified.The modified file is at: src\obj_iotdk_10\gnu_arcem9d\linker_gnu.ldf

### CODE REVIEW

**Comment by hustmzp** [comment]
The link script has been modified.The modified file is at: src\obj_iotdk_10\gnu_arcem9d\linker_gnu.ldf

File: arc_design_contest/2019/HUST_SoundSnake/src/appl_mem_config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#define REGION_MLI_DATA		REGION_DCCM
+#define REGION_MLI_BSS		REGION_DCCM
+#define REGION_MLI_ZDATA	REGION_DCCM
+#define REGION_MLI_MODEL_P2	REGION_DCCM
+#define REGION_MLI_MODEL	REGION_XCCM
+#define REGION_MLI_XDATA	REGION_XCCM
+#define REGION_MLI_YDATA	REGION_YCCM
+#endif
+
+#endif /* _APPL_MEM_CONFIG_H_ */
```
Comment (IRISZZW) [inline arc_design_contest/2019/HUST_SoundSnake/src/appl_mem_config.h]: file end with a new line

File: arc_design_contest/2019/HUST_SoundSnake/src/codec.h [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#ifndef _CODEC_H_
+#define _CODEC_H_
+
+#include "embARC.h"
+#include "embARC_debug.h"
+#include "wm8978i2s.h"
+
+void CODEC_init(void);
+#endif
```
Comment (IRISZZW) [inline arc_design_contest/2019/HUST_SoundSnake/src/codec.h]: end new line

File: arc_design_contest/2019/HUST_SoundSnake/src/kws.h [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+#define FINAL_FC_OUT_RSHIFT 7
+
+#define SCRATCH_BUFFER_SIZE (2*(2*64*3*3+64*5*25))
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+/*
+ * @brief cnn相关初始化
```
Comment (IRISZZW) [inline arc_design_contest/2019/HUST_SoundSnake/src/kws.h]: please translate all Chinese comments to English

File: arc_design_contest/2019/HUST_SoundSnake/src/makefile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# Application name
+APPL ?= SoundSnake
+
+#APPL_DEFINES = -DUSE_APPL_MEM_CONFIG -DV2DSP_XY -DMODEL_BIT_DEPTH=8
+
+#
+# root dir of embARC
+#
+EMBARC_ROOT = ../../..
```
Comment (IRISZZW) [inline arc_design_contest/2019/HUST_SoundSnake/src/makefile]: project folder should be embarc_osp\embarc_applications\arc_design_contest\2019\HUST_SoundSnake\src

so the EMBARC_ROOT should be ../../../../..

**Pull request merged**